docs/examples/files.md

```diff
@@ -0,0 +1,231 @@
+    support for parsing this type of data.
+
+## JSON data
+
+`.json` files are a common way to store key / value data in a human-readable format.
+Here is an example of a `.json` file:
+
+```json
+{
+    "name": "John Doe",
+    "age": 30,
+    "email": "john@example.com"
+}
+```
+
+To validate this data, we can use a `pydantic` model:
+
+```python test="skip"
+from pydantic import BaseModel, EmailStr, PositiveInt
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+with open('person.json') as f:
+    json_string = f.read()
+    person = Person.model_validate_json(json_string)
+    print(repr(person))
+    #> Person(name='John Doe', age=30, email='john@example.com')
+```
+
+If the data in the file is not valid, `pydantic` will raise a [`ValidationError`][pydantic_core.ValidationError].
+Let's say we have the following `.json` file:
+
+```json
+{
+    "age": -30,
+    "email": "not-an-email-address"
+}
+```
+
+This data is flawed for three reasons:
+1. It's missing the `name` field.
+2. The `age` field is negative.
+3. The `email` field is not a valid email address.
+
+When we try to validate this data, `pydantic` raises a [`ValidationError`][pydantic_core.ValidationError] with all of the
+above issues:
+
+```python test="skip"
+from pydantic import BaseModel, EmailStr, PositiveInt, ValidationError
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+with open('person.json') as f:
+    json_string = f.read()
+    try:
+        person = Person.model_validate_json(json_string)
+    except ValidationError as err:
+        print(err)
+        """
+        3 validation errors for Person
+        name
+        Field required [type=missing, input_value={'age': -30, 'email': 'not-an-email-address'}, input_type=dict]
+            For further information visit https://errors.pydantic.dev/2.10/v/missing
+        age
+        Input should be greater than 0 [type=greater_than, input_value=-30, input_type=int]
+            For further information visit https://errors.pydantic.dev/2.10/v/greater_than
+        email
+        value is not a valid email address: An email address must have an @-sign. [type=value_error, input_value='not-an-email-address', input_type=str]
+        """
+```
+
+Often, it's the case that you have an abundance of a certain type of data within a `.json` file.
+For example, you might have a list of people:
+
+```json
+[
+    {
+        "name": "John Doe",
+        "age": 30,
+        "email": "john@example.com"
+    },
+    {
+        "name": "Jane Doe",
+        "age": 25,
+        "email": "jane@example.com"
+    }
+]
+```
+
+In this case, you can validate the data against a `List[Person]` model:
+
+```python test="skip"
+from typing import List
+
+from pydantic import BaseModel, EmailStr, PositiveInt, TypeAdapter
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+type_adapter = TypeAdapter(List[Person])  # (1)!
```

Comment:
```suggestion
person_list_adapter = TypeAdapter(List[Person])  # (1)!
```
I think better to name the variable to reflect the type it handles, that's a pattern I'd want to see in my own code bases, so doing it in the docs is a good way to encourage users to do the same, imo.